qiskit_algorithms/gradients/reverse/reverse_qgt.py

```diff
@@ -81,21 +82,21 @@ def _run(  # pylint: disable=arguments-renamed
         self,
         circuits: Sequence[QuantumCircuit],
         parameter_values: Sequence[Sequence[float]],
-        parameter_sets: Sequence[set[Parameter]],
+        parameters: Sequence[Sequence[Parameter]],
         **options,
     ) -> QGTResult:
         """Compute the QGT on the given circuits."""
         g_circuits, g_parameter_values, g_parameter_sets = self._preprocess(
-            circuits, parameter_values, parameter_sets, self.SUPPORTED_GATES
+            circuits, parameter_values, parameters, self.SUPPORTED_GATES
         )
         results = self._run_unique(g_circuits, g_parameter_values, g_parameter_sets, **options)
-        return self._postprocess(results, circuits, parameter_values, parameter_sets)
+        return self._postprocess(results, circuits, parameter_values, parameters)
 
     def _run_unique(
         self,
         circuits: Sequence[QuantumCircuit],
         parameter_values: Sequence[Sequence[float]],
-        parameter_sets: Sequence[set[Parameter]],
+        parameter_sets: Sequence[Sequence[Parameter]],
```

Comment:
Seeing this again here I finally commented maybe this needs further investigation. Given its called parameter_sets and a set having unique values in it, unlike Sequence - maybe it has never really been treated as a set or needed to be one. If it were assumed so and this now has duplicates, which maybe a test never covered (CI passes). @a-matsuo any comment/thought here.